PasswordRules.cs

```diff
@@ -0,0 +1,151 @@
+            }
+            if (!ContainsSymbol(pass))
+            {
+                return false;
+            }
+            if (!ContainsDigit(pass))
+            {
+                return false;
+            }
+            if (!ContainsLowerLetter(pass))
+            {
+                return false;
+            }
+            if (!ContainsUpperLetter(pass))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public void CheckPasswordByApi(string pass)
+        {
+            string sHash = string.Empty;
+
+            using (SHA1Managed sha1 = new SHA1Managed()) // use the class to hash the password 
+            {
+                var hash = sha1.ComputeHash(Encoding.UTF8.GetBytes(pass)); // create the hash as a bit array
+                sHash = BitConverter.ToString(hash).Replace("-", ""); // convert bit array to hexadecimal
+            }
+
+            string existingPasswords = requestThroughApi(sHash.Substring(0, 5)); //call the function for API request
+            if (!string.IsNullOrWhiteSpace(existingPasswords)) // if string returned, we will check our password
+            {
+                var passwordAr = existingPasswords.Split("\r\n", StringSplitOptions.None).ToList(); //splitting the string into separate passwords
+
+                Dictionary<string, int> passwordDic = new Dictionary<string, int>(); // create a dictionary to collect password and for quantity of how often does the password appear in API
+                foreach (string pw in passwordAr)
+                {
+                    var tmpPw = pw.Split(':'); //split the string on password and appearing times
+                    passwordDic.TryAdd(tmpPw[0], int.Parse(tmpPw[1])); //adding to dictionary
+                }
+
+                int occassions = 0;
+                passwordDic.TryGetValue(sHash, out occassions); //try to find quantity of password appearing times
+                if (occassions == 0) // if there are not appearing times, the password is good
+                {
+                    Console.WriteLine("Password is good!");
+                }
+            }
+            else
+            {
+                Console.WriteLine("Your password appeared in Database, so it is no good");
+            }
+        }
+        private static string requestThroughApi(string hash)
+        {
+            string result = string.Empty; //result string
+
+            var url = "https://api.pwnedpasswords.com/range/" + hash;// where  to make a request
+
+            var httpRequest = (HttpWebRequest)WebRequest.Create(url); // create and obj http request
+            httpRequest.Method = "GET"; // choose the method, which needed to us
+
+            var httpResponse = (HttpWebResponse)httpRequest.GetResponse(); // try to make a request
+            using (var streamReader = new StreamReader(httpResponse.GetResponseStream())) // create and object StreamReader to read the result

```

Comment:
To avoid errors we could wrap this with a small check:
` if (httpResponse.StatusCode == HttpStatusCode.OK)`

(Checks if the response returned HTTP Status Code 200 (Success)